### DISCUSSION TITLE
Switch on/off/toggle message without triggering messages

### DISCUSSION THREAD

**Discussion by GriffinSauce** [Pirate-MIDI/Pirate-MIDI-Features-Bug-Tracking#86] [Feature Requests & Ideas]
Right now when a "Switch On" smart message is executed, any midi messages related to "Toggle on" for that switch will be sent.

It would be very helpful to have a "Without sending messages" option with the Switch on/off/toggle smart messages to only change the state of the switch on the Bridge6. This would make complex configurations and interactions more easy to manage.

For example in my config:
- I have a loop switcher that only responds to PC messages with drive pedals in loops 1 and 2 that I only want to engage individually
- Switch 5 is configured to switch loop 1 and 2 on the looper on the primary and secondary switches respectively
- Pressing Switch 3 (rhythm preset) toggles "Primary off" and "Secondary on"
- This means the PC message for "all loops off" and "loop 2 on" will be sent very consecutively very quickly and cause sound issues with the looper. It appears that it even messes with the internal state management because it can't handle switching that fast.

There is no way to toggle only the individual state with a program-based loop switcher. So I would like to just change the state only for Switch 5 and send a single PC message from Switch 3 to the looper directly.

It's a bit of a convoluted example because you'd want to switch loops with a CC message and it would work fine to engage/disengage simultaneously but not all loop switchers or relays support that. (example: Lehle D.Loop, Nux PMS2). I've also encountered other examples in the past while working with my Gordius Little Giant where this comes in handy.

**Comment by simonaglover95** [comment]
Yeah, I can see that being useful. We've had a couple of people mention that this might be a useful feature for them.
